documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -210,41 +210,43 @@ These are relatively simple list wrappers with the events fired when state chang
 public partial class Mice : IReadOnlyList<IMouse>
 {
     public MouseClickConfiguration ClickConfiguration { get; set; }
-    public event Action<ButtonEvent<IMouse, MouseButton>>? ButtonDown;
-    public event Action<ButtonEvent<IMouse, MouseButton>>? ButtonUp;
-    public event Action<ClickEvent<IMouse, MouseButton>>? Click;
-    public event Action<ClickEvent<IMouse, MouseButton>>? DoubleClick;
-    public event Action<AxisEvent<IMouse, Vector2>>? CursorMove;
-    public event Action<AxisEvent<IMouse, Vector2>>? Scroll;
+    public event Action<MouseDownEvent>? ButtonDown;
+    public event Action<MouseUpEvent>? ButtonUp;
+    public event Action<MouseClickEvent>? Click;
+    public event Action<MouseClickEvent>? DoubleClick;
+    public event Action<MouseMoveEvent>? CursorMove;
+    public event Action<MouseScrollEvent>? Scroll;
 }
 
 public partial class Keyboards : IReadOnlyList<IKeyboard>
 {
-    public event Action<ButtonEvent<IKeyboard, Key>>? KeyDown;
-    public event Action<ButtonEvent<IKeyboard, Key>>? KeyUp;
-    public event Action<TextEvent<IKeyboard>>? TextInput;
+    public event Action<KeyDownEvent>? KeyDown;
+    public event Action<KeyUpEvent>? KeyUp;
+    public event Action<KeyTextEvent>? TextInput;
```

Comment:
I still believe we should expose `KeyChar` that works like in 2.X. Specially now with the `IInputHandler` model it wouldn't make sense to keep this unless we added logic to make sure all the `HandleKeyChar()`-s  are merged into a single `TextInput` event.

What if for example, someone wants to make a typing game? Handling text input like this makes it easier for the specific case in which the user doesn't need to filter chars and _needs_ it to be a string (for my text input UI, I always have a `StringBuilder` or something more performant) but makes every other use case harder

Reply:
Removing the text input event to ensure the events match their handlers, but if they want the full text it will still be available on the state if within BeginInput/EndInput calls.

```suggestion
    public event Action<KeyCharEvent>? KeyChar;
```